pennylane/transforms/decompositions/two_qubit_unitary.py

```diff
@@ -66,38 +54,45 @@ def _convert_to_su4(U):
     # Convert to SU(4) if it's not close to 1
     if not math.allclose(det, 1.0):
         exp_angle = -1j * math.cast_like(math.angle(det), 1j) / 4
-        U = math.cast_like(U, det) * qml.math.exp(exp_angle)
+        U = math.cast_like(U, det) * math.exp(exp_angle)
 
     return U
 
 
-def _select_rotation_angles(U):
-    r"""Choose the rotation angles of RZ, RY in the two-qubit decomposition.
-    They are chosen as per Proposition V.1 in quant-ph/0308033 and are based
-    on the phases of the eigenvalues of :math:`E^\dagger \gamma(U) E`, where
+def _compute_num_cnots(U):
+    r"""Compute the number of CNOTs required to implement U. This is based on
+    the trace of
 
     .. math::
 
-        \gamma(U) = (E^\dag U E) (E^\dag U E)^T.
+        \gamma(U) = (E^\dag U E) (E^\dag U E)^T,
+
+    and follows the arguments of this paper: https://arxiv.org/abs/quant-ph/0308045
     """
+    u = math.dot(Edag, math.dot(U, E))
+    gammaU = math.dot(u, math.T(u))
 
-    gammaU = qml.math.linalg.multi_dot([Edag, U, E, Et, qml.math.T(U), qml.math.T(Edag)])
-    evs = qml.math.linalg.eigvals(gammaU)
+    trace = qml.math.trace(gammaU)
```

Comment:
how come it is just `math` in some places, and `qml.math` in others? (just curious)

Reply:
Ahh I had at one point changed everything to just `math`, but was not consistent when I added more stuff. I'll do an update, thanks for catching this! :+1: